pkg/controller/replication_controller.go

```diff
@@ -351,16 +351,20 @@ func (rm *ReplicationManager) syncReplicationController(key string) error {
 	}
 	controller := *obj.(*api.ReplicationController)
 
+	// Check the expectations of the rc before counting active pods, otherwise a new pod can sneak in
+	// and update the expectations after we've retrieved active pods from the store. If a new pod enters
+	// the store after we've checked the expectation, the rc sync is just deferred till the next relist.
```

Comment:
We are defenseless against a pod sneaking in, it's only bad if it does so after we've counted the active pods but before we've checked the expectation. Previously there was a race:
expectations = 1
we count active pods = 1
-- 2nd pod arrives, sets expectations = 0 
we check that expectations are satisfied and pass 1 active pods to manageReplicas, which creates a third pod

Now, if the expectation was 1 when we checked it, even if the pod sneaked in we won't invoke manageReplicas.